### PR TITLE
retry on AWS LimitExceededException error for kinesis

### PR DESCRIPTION
**Pull request by mechpen**

### CODE REVIEW

File: c7n/query.py [hunk 1]
```diff
@@ -361,6 +361,7 @@ class QueryResourceManager(ResourceManager):
         get_retry((
             'ThrottlingException',
             'RequestLimitExceeded',
+            'LimitExceededException',
```
Comment (thisisshi) [inline c7n/query.py]: I don't think we should be retrying on this error as this is typically a response from AWS that indicates that your account has reached a maximum limit of a certain resource, AWS's API response behavior is somewhat inconsistent on this front, as documented here: https://github.com/aws/aws-sdk-go/issues/1376

I'm not sure how exactly we can get around this other than defining custom retryable errors in the class definition of the resource itself to make sure we aren't retrying when we shouldn't be.

Comment (mechpen) [reply c7n/query.py]: Thanks for the link.  This explains why our read only kinesis policy fails with LimitExceededException.  I'll see if I could make a better fix.

**Comment by mechpen** [comment]
I moved the retry error code into KinesisStream class.

File: c7n/resources/kinesis.py [hunk 2]
```diff
@@ -17,11 +17,14 @@
...
 from c7n.tags import universal_augment
-from c7n.utils import local_session, type_schema
+from c7n.utils import local_session, type_schema, get_retry
 
 
 @resources.register('kinesis')
 class KinesisStream(QueryResourceManager):
+    retry = staticmethod(
+        get_retry((
+            'LimitExceededException')))
```
Comment (kapilt) [inline c7n/resources/kinesis.py]: this needs a trailing comma, else its just passing the string through.

```python
>>> x = ('foo')
>>> x
'foo'
>>> x = ('foo',)
>>>
```

Comment (mechpen) [reply c7n/resources/kinesis.py]: fixed

**Pull request merged**